api/ttn/lorawan/v3/qrcodegenerator.proto

```diff
@@ -130,3 +131,38 @@ service EndDeviceQRCodeGenerator {
     };
   }
 }
+
+message ParseGatewayQRCodeRequest {
+  // QR code format identifier.
+  // If this field is not specified, the server will default to ttigpro1.
+  string format_id = 1 [(validate.rules).string = {
+    pattern: "^[a-z0-9](?:[-]?[a-z0-9]){2,}$|^$",
+    max_len: 36
+  }];
+  // Raw QR code contents.
+  bytes qr_code = 2 [(validate.rules).bytes = {
+    min_len: 10,
+    max_len: 1024
+  }];
+}
+
+message ParseGatewayQRCodeResponse {
+  // Identifier of the format used to successfully parse the QR code data.
+  string format_id = 1;
+  ClaimGatewayRequest claim_gateway_request = 2;
```

Comment:
Sorry for raising this only now but this tightly couples QR code scanning with claiming.

QR codes can be numerous things, not just claiming. For example, the EUI only, for identification.

Lacking the equivalent of `EndDeviceTemplate` for gateways, I would return fields here that are included in gateway QR codes, starting with the gateway EUI and owner token.

Reply:
done